### PR TITLE
IBX-2589: Added new commands for SearchField for EmailAddress and added transformByGroup lowercase for index

### PR DESCRIPTION
**Pull request by mateuszdebinski**
| Question                                  | Answer
| ---------------------------------------- | ------------------
| **JIRA issue**                          | [IBX-2589](https://issues.ibexa.co/browse/IBX-2589)
| **Type**                                   | bug
| **Target eZ Platform version** | `v2.5`
| **BC breaks**                          | no

#### Checklist:
- [x] Provided PR description.
- [x] Tested the solution manually.
- [x] Provided automated test coverage.
- [x] Checked that target branch is set correctly (master for features, the oldest supported for bugs).
- [x] Ran PHP CS Fixer for new PHP code (use `$ composer fix-cs`).
- [x] Asked for a review (ping `@ezsystems/engineering-team`).

### CODE REVIEW

File: eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php [hunk 1]
```diff
@@ -854,6 +856,48 @@ public function testIndexingSpecialFullTextCases($text, $searchForText, array $i
         $this->fail('Failed to find required Content in search results');
     }
 
+    /**
+     * Check if FullText indexing works for email addresses.
+     *
+     * @dataProvider getEmailAddressesCases
+     */
+    public function testIndexingEmailFieldCases(string $email, string $searchForText)
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php]:
```suggestion
    public function testIndexingEmailFieldCases(string $email, string $searchForText): void
```

File: eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php [hunk 2]
```diff
@@ -854,6 +856,48 @@ public function testIndexingSpecialFullTextCases($text, $searchForText, array $i
...
+        }
+        $this->fail('Failed to find required Content in search results');
+    }
+
+    /**
+     * Data Provider for {@see testIndexingSpecialFullTextCases()} method.
+     *
+     * @return array
+     */
+    public function getEmailAddressesCases()
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php]:
```suggestion
    public function getEmailAddressesCases(): array
```

**Review by adamwojs — APPROVED** [review]

File: eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php [hunk 3]
```diff
@@ -854,6 +856,48 @@ public function testIndexingSpecialFullTextCases($text, $searchForText, array $i
...
+        $query = new Query(['filter' => $criterion]);
+        $result = $searchService->findContent($query);
+
+        // for some cases there might be more than one hit, so check if proper one was found
+        foreach ($result->searchHits as $searchHit) {
+            if ($content->contentInfo->id === $searchHit->valueObject->versionInfo->contentInfo->id) {
+                return;
+            }
+        }
+        $this->fail('Failed to find required Content in search results');
```
Comment (adamwojs) [inline eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php]: Could you please change $this->fail(...) to assertion?  

```php
	$found = false;
        foreach ($result->searchHits as $searchHit) {
            if ($content->contentInfo->id === $searchHit->valueObject->versionInfo->contentInfo->id) {
                $found = true;
                break;
            }
        }
        
        
        self::assertTrue($found, 'Failed to find required Content in search results');
```

Comment (mateuszdebinski) [reply eZ/Publish/API/Repository/Tests/SearchEngineIndexingTest.php]: I changed this into two functions testIndexingEmailFieldCases and testIndexingSpecialFullTextCases

**Review by mikadamczyk — APPROVED** [review]

**Review by bogusez — APPROVED** [review]

**Review by Steveb-p — APPROVED** [review]

**Pull request merged**